plugins/aws/sts_provisioner.go

```diff
@@ -97,9 +92,14 @@ func (p STSProvisioner) Description() string {
 }
 
 // ChooseTemporaryCredentialsProvider returns the aws provider that fits the scenario described by the current configuration.
-func ChooseTemporaryCredentialsProvider(awsConfig *confighelpers.Config, providerFactory STSProviderFactory) (aws.CredentialsProvider, error) {
+func ChooseTemporaryCredentialsProvider(awsConfig *confighelpers.Config, providerFactory STSProviderFactory, itemFields map[sdk.FieldName]string) (aws.CredentialsProvider, error) {
+	err := resolveLocalAnd1PasswordConfigurations(itemFields, awsConfig)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
what is the reason this check is now required, but it was not before?

Reply:
This was required before as well, I just moved it over from another function inside here, so it would be performed for each profile, because now profiles can look into each other recursively, by calling function `ChooseTemporaryCredentialsProvider ` recursively. Before it was called in the caller function of `ChooseTemporaryCredentialsProvider` so it would resolve fields only for first profile in the chain.